src/internal.c

```diff
@@ -4280,6 +4320,10 @@ void FreeHandshakeHashes(WOLFSSL* ssl)
     #ifdef WOLFSSL_SHA512
         wc_Sha512Free(&ssl->hsHashes->hashSha512);
     #endif
+    #if defined(HAVE_ED25519) && !defined(WOLFSSL_NO_CLIENT_AUTH)
+        if (ssl->hsHashes->messages != NULL)
+            XFREE(ssl->hsHashes->messages, ssl->heap, DYNAMIC_TYPE_HASHES);
```

Comment:
The pattern here is to set the pointer to NULL after XFREE as in the next couple lines.  This allows early resource free after a good handshake but also allows proper cleanup w/o double freeing at the end of the WOLFSSL lifetime.

Reply:
hsHashes is being freed explicitly directtly after. Will make change as specified.